kibbeh/src/ui/RoomCard.tsx

```diff
@@ -0,0 +1,76 @@
+import { format, isToday, isPast, differenceInMilliseconds } from "date-fns";
+import React, { useEffect, useState } from "react";
+import { SmSolidTime } from "../icons";
+import { BubbleText } from "./BubbleText";
+import { RoomCardHeading } from "./RoomCardHeading";
+import { Tag } from "./Tag";
+
+function formatNumber(num: number): string {
+  return Math.abs(num) > 999
+    ? `${Math.sign(num) * Number((Math.abs(num) / 1000).toFixed(1))}K`
+    : `${Math.sign(num) * Math.abs(num)}`;
+}
+
+function useScheduleRerender(scheduledFor: Date) {
+  // same logic stolen from kofta, rerenders
+  // at the scheduleFor date
```

Comment:
What if the host isn't online when the room starts? I'd probably leave a 15min margin before deleting/hiding the room instead of just going live directly, which might look weird and confusing. Let's wait for @benawad 's opinion.

Reply:
It doesn't hide anything right now, it only switches state from scheduled to live. As it's probably going to be used in `feed` it wouldn't make sense to `delete/hide` the card but make the room accessible perhaps? Or maybe people could access the room while waiting for the host, similar to YouTube lives where people can start chat while the host is starting.  But indeed it require more clarification on how it's going to work.

Reply:
Then when that happens maybe change the time for `Soon`? But for me what would make sense is that the host had a `Go Live` button.

Being able to chat prior to the room starting would be interesting. Like being able to enter the room, having all the listeners and then having the start time in the speakers section.

Reply:
I'll probably be changing when exactly it rerenders but that's an easy tweak we can do later.

The mockup is missing some actions for scheduled cards i.e. go live, edit, delete, add to calendar, # of people attending, cohosts and copy link

We can come back and edit `RoomCard` after we know what scheduled card looks like

Reply:
Oh I see, so this component is more like a `FeedCard` then. Anyhow I think we can resolve this for now. I'll push the other fixes.